vendor/k8s.io/kubernetes/pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -438,8 +438,12 @@ func (e *EndpointController) syncService(key string) error {
 	readyEps := 0
 	notReadyEps := 0
 	for i := range pods {
-		// TODO: Do we need to copy here?
-		pod := &(*pods[i])
+		podObj, err := api.Scheme.DeepCopy(pods[i])
```

Comment:
This fix doesn't look right.  I'm not seeing a mutation on the pod.  You need to find the mutator, not chase all other accesses.